newrelic_lambda_cli/integrations.py

```diff
@@ -77,74 +77,64 @@ def create_role(session, role_policy, nr_account_id):
+        ParametersOverrides=parameters,
+        Capabilities=capabilities
     )
-    with open(template_path) as template:
-        client.create_stack(
-            StackName=stack_name,
-            TemplateBody=template.read(),
-            Parameters=[
-                {"ParameterKey": "MemorySize", "ParameterValue": str(memory_size)},
-                {
-                    "ParameterKey": "NewRelicLicenseKey",
-                    "ParameterValue": nr_license_key,
-                },
-                {
-                    "ParameterKey": "NewRelicLoggingEnabled",
-                    "ParameterValue": "True" if enable_logs else "False",
-                },
-                {"ParameterKey": "Timeout", "ParameterValue": str(timeout)},
-            ],
-            Capabilities=["CAPABILITY_NAMED_IAM", "CAPABILITY_AUTO_EXPAND"],
-        )
-        click.echo(
-            "Waiting for stack creation to complete, this may take a minute... ",
-            nl=False,
-        )
-        client.get_waiter("stack_create_complete").wait(StackName=stack_name)
-        success("Done")
+
+    cf_client = session.client("cloudformation")
+    cf_client.execute_change_set(
+        ChangeSetName = change_set['ChangeSetId']
+    )
+
+    click.echo(
+        "Waiting for stack creation to complete, this may take a minute... ",
+        nl=False,
+    )
+    client.get_waiter("stack_create_complete").wait(StackName=stack_name)
+    success("Done")
 
 
 def update_log_ingestion_function(
-    session, nr_license_key, enable_logs=False, memory_size=128, timeout=30
+    session, nr_license_key, enable_logs=False, memory_size=128, timeout=30, role_name=None
 ):
-    client = session.client("cloudformation")
-    stack_name = "NewRelicLogIngestion"
-    template_path = os.path.join(
-        os.path.dirname(os.path.abspath(__file__)),
-        "templates",
-        "newrelic-log-ingestion.yaml",
-    )
-    with open(template_path) as template:
-        client.update_stack(
-            StackName=stack_name,
-            TemplateBody=template.read(),
-            Parameters=[
-                {"ParameterKey": "MemorySize", "ParameterValue": str(memory_size)},
-                {
-                    "ParameterKey": "NewRelicLicenseKey",
-                    "ParameterValue": nr_license_key,
-                },
-                {
-                    "ParameterKey": "NewRelicLoggingEnabled",
-                    "ParameterValue": "True" if enable_logs else "False",
-                },
-                {"ParameterKey": "Timeout", "ParameterValue": str(timeout)},
-            ],
-            Capabilities=["CAPABILITY_NAMED_IAM", "CAPABILITY_AUTO_EXPAND"],
-        )
-        click.echo(
-            "Waiting for stack update to complete, this may take a minute... ", nl=False
-        )
-        client.get_waiter("stack_update_complete").wait(StackName=stack_name)
-        success("Done")
+    # TODO: we need to detect an old-style nested install and unwrap it
```

Comment:
I think we should address this before merging this change to ensure a clean update path.

Reply:
Definitely.